code/obj/item/rcd.dm

```diff
@@ -190,10 +185,15 @@ Broken RCD + Effects
 				boutput(user, "This cartridge is not made of the proper material to be used in \The [src].")
 
 	attack_self(mob/user as mob)
+		user.showContextActions(src.contextActions, src, src.contextLayout)
+
+	proc/switch_mode(var/mode, var/mob/user)
+		if (!(mode in src.modes))
+			return
```

Comment:
This should probably `CRASH` with some debug info- we want to know if something is passing nonsense as an RCD mode (or maybe someone just made a typo, etc)